test/parallel/test-fs-write-optional-params.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const common = require('../common');
+
+// This test ensures that fs.write accepts "named parameters" object
+// and doesn't interpret objects as strings
+
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+const tmpdir = require('../common/tmpdir');
+const util = require('util');
+
+tmpdir.refresh();
+
+const destInvalid = path.resolve(tmpdir.path, 'rwopt_invalid');
+const buffer = Buffer.from('zyx');
+
+const close = util.promisify(fs.close);
+const open = util.promisify(fs.open);
+const read = util.promisify(fs.read);
+const write = util.promisify(fs.write);
```

Comment:
Can you instead change the `testInvalid` and `testValid` signatures to accept a callback, and promisify them?

Reply:
Yep, at the cost of `...restParams` in `testInvalid()`. Not a big deal, but are there non-stylistic reasons for that change?

Reply:
We want to test the actual fs methods, not their promisified version.